plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostModelInterface.hs

```diff
@@ -101,20 +102,32 @@ data CostModelApplyError =
       -- ^ internal error when we are transforming the applyParams' input to json (should not happen)
     | CMInternalWriteError String
       -- ^ internal error when we are transforming the applied params from json with given jsonstring error (should not happen)
-    | CMWrongNumberOfParams Int Int
-      -- ^ the ledger is supposed to pass the full list of params, no more, no less params.
+    | CMTooFewParamsError Int Int
+      -- ^ the ledger is supposed to pass the exact number of expected params,
+      -- or more params in the case of out-of-date node. The ledger should never pass two few params.
```

Comment:
I think this is fiddly enough that we should promote it to a Note about what our expectations are about the parameters the ledger gives us, and then just reference it here. There was quite a lot we discussed, so let's write it down!